src/pages/VeSDL/OnChainVote.tsx

```diff
@@ -0,0 +1,254 @@
+import {
+  Alert,
+  Autocomplete,
+  Box,
+  Button,
+  Link,
+  Stack,
+  TextField,
+  Typography,
+  useTheme,
+} from "@mui/material"
+import React, { useCallback, useEffect, useMemo, useState } from "react"
+import { commify, formatBNToString, isNumberOrEmpty } from "../../utils"
+import ArrowDownIcon from "@mui/icons-material/KeyboardArrowDown"
+import { BigNumber } from "ethers"
+import { Gauge } from "../../utils/gauges"
+import VoteHistory from "./Votes"
+import { enqueuePromiseToast } from "../../components/Toastify"
+import { useActiveWeb3React } from "../../hooks"
+import { useGaugeControllerContract } from "../../hooks/useContract"
+import { useTranslation } from "react-i18next"
+
+interface OnChainVoteProps {
+  veSdlBalance: BigNumber
+  gauges: Partial<{
+    [lpTokenAddress: string]: Gauge
+  }>
+}
+export type GaugeName = {
+  address: string
+  gaugeName: string
+}
+
+export type VotesType = {
+  [gaugeAddress: string]: {
+    gaugeName: string
+    weight: BigNumber
+    voteDate: number
+  }
+}
+
+export default function OnChainVote({
+  veSdlBalance,
+  gauges,
+}: OnChainVoteProps) {
+  const theme = useTheme()
+  const { account } = useActiveWeb3React()
+  const { t } = useTranslation()
+  const [selectedGauge, setSelectedGauge] = useState<GaugeName | null>(null)
+  const [alertMessage, setAlertMessage] = useState<string>()
+  const [voteWeightToSubmit, setVoteWeightToSubmit] = useState<string>("")
+  const [voteUsed, setVoteUsed] = useState<number | undefined>()
+  const [votes, setVotes] = useState<VotesType>({})
+  const gaugeControllerContract = useGaugeControllerContract(true)
+  const [voteWeightInputError, setVoteWeightInputError] = useState<{
+    hasError: boolean
+    errorText: string
+  }>({ hasError: false, errorText: " " })
+
+  const gaugeNames = useMemo(
+    () =>
+      Object.keys(gauges)
+        .filter((lpAddress) => !gauges[lpAddress]?.isKilled)
+        .map((lpAddress) => {
+          const gaugeName = gauges[lpAddress]?.gaugeName
```

Comment:
can you early null check/return gauges[lpAddress].<property> so the value can't be undefined during this execution?